// bootstrap-vue/es/components/plugins.js
"use strict";

exports.__esModule = true;
exports.TooltipPlugin = exports.ToastPlugin = exports.TabsPlugin = exports.TablePlugin = exports.SpinnerPlugin = exports.ProgressPlugin = exports.PopoverPlugin = exports.PaginationNavPlugin = exports.PaginationPlugin = exports.NavbarPlugin = exports.NavPlugin = exports.ModalPlugin = exports.MediaPlugin = exports.ListGroupPlugin = exports.LinkPlugin = exports.JumbotronPlugin = exports.ImagePlugin = exports.FormSelectPlugin = exports.FormFilePlugin = exports.FormTextareaPlugin = exports.FormInputPlugin = exports.FormRadioPlugin = exports.FormCheckboxPlugin = exports.FormGroupPlugin = exports.FormPlugin = exports.EmbedPlugin = exports.DropdownPlugin = exports.CollapsePlugin = exports.LayoutPlugin = exports.CarouselPlugin = exports.CardPlugin = exports.InputGroupPlugin = exports.ButtonToolbarPlugin = exports.ButtonGroupPlugin = exports.ButtonPlugin = exports.BreadcrumbPlugin = exports.BadgePlugin = exports.AlertPlugin = exports.BVToastPlugin = exports.BVModalPlugin = void 0;

var _bvModal = _interopRequireDefault(require("./modal/helpers/bv-modal"));

exports.BVModalPlugin = _bvModal.default;

var _bvToast = _interopRequireDefault(require("./toast/helpers/bv-toast"));

exports.BVToastPlugin = _bvToast.default;

var _alert = _interopRequireDefault(require("./alert"));

exports.AlertPlugin = _alert.default;

var _badge = _interopRequireDefault(require("./badge"));

exports.BadgePlugin = _badge.default;

var _breadcrumb = _interopRequireDefault(require("./breadcrumb"));

exports.BreadcrumbPlugin = _breadcrumb.default;

var _button = _interopRequireDefault(require("./button"));

exports.ButtonPlugin = _button.default;

var _buttonGroup = _interopRequireDefault(require("./button-group"));

exports.ButtonGroupPlugin = _buttonGroup.default;

var _buttonToolbar = _interopRequireDefault(require("./button-toolbar"));

exports.ButtonToolbarPlugin = _buttonToolbar.default;

var _inputGroup = _interopRequireDefault(require("./input-group"));

exports.InputGroupPlugin = _inputGroup.default;

var _card = _interopRequireDefault(require("./card"));

exports.CardPlugin = _card.default;

var _carousel = _interopRequireDefault(require("./carousel"));

exports.CarouselPlugin = _carousel.default;

var _layout = _interopRequireDefault(require("./layout"));

exports.LayoutPlugin = _layout.default;

var _collapse = _interopRequireDefault(require("./collapse"));

exports.CollapsePlugin = _collapse.default;

var _dropdown = _interopRequireDefault(require("./dropdown"));

exports.DropdownPlugin = _dropdown.default;

var _embed = _interopRequireDefault(require("./embed"));

exports.EmbedPlugin = _embed.default;

var _form = _interopRequireDefault(require("./form"));

exports.FormPlugin = _form.default;

var _formGroup = _interopRequireDefault(require("./form-group"));

exports.FormGroupPlugin = _formGroup.default;

var _formCheckbox = _interopRequireDefault(require("./form-checkbox"));

exports.FormCheckboxPlugin = _formCheckbox.default;

var _formRadio = _interopRequireDefault(require("./form-radio"));

exports.FormRadioPlugin = _formRadio.default;

var _formInput = _interopRequireDefault(require("./form-input"));

exports.FormInputPlugin = _formInput.default;

var _formTextarea = _interopRequireDefault(require("./form-textarea"));

exports.FormTextareaPlugin = _formTextarea.default;

var _formFile = _interopRequireDefault(require("./form-file"));

exports.FormFilePlugin = _formFile.default;

var _formSelect = _interopRequireDefault(require("./form-select"));

exports.FormSelectPlugin = _formSelect.default;

var _image = _interopRequireDefault(require("./image"));

exports.ImagePlugin = _image.default;

var _jumbotron = _interopRequireDefault(require("./jumbotron"));

exports.JumbotronPlugin = _jumbotron.default;

var _link = _interopRequireDefault(require("./link"));

exports.LinkPlugin = _link.default;

var _listGroup = _interopRequireDefault(require("./list-group"));

exports.ListGroupPlugin = _listGroup.default;

var _media = _interopRequireDefault(require("./media"));

exports.MediaPlugin = _media.default;

var _modal = _interopRequireDefault(require("./modal"));

exports.ModalPlugin = _modal.default;

var _nav = _interopRequireDefault(require("./nav"));

exports.NavPlugin = _nav.default;

var _navbar = _interopRequireDefault(require("./navbar"));

exports.NavbarPlugin = _navbar.default;

var _pagination = _interopRequireDefault(require("./pagination"));

exports.PaginationPlugin = _pagination.default;

var _paginationNav = _interopRequireDefault(require("./pagination-nav"));

exports.PaginationNavPlugin = _paginationNav.default;

var _popover = _interopRequireDefault(require("./popover"));

exports.PopoverPlugin = _popover.default;

var _progress = _interopRequireDefault(require("./progress"));

exports.ProgressPlugin = _progress.default;

var _spinner = _interopRequireDefault(require("./spinner"));

exports.SpinnerPlugin = _spinner.default;

var _table = _interopRequireDefault(require("./table"));

exports.TablePlugin = _table.default;

var _tabs = _interopRequireDefault(require("./tabs"));

exports.TabsPlugin = _tabs.default;

var _toast = _interopRequireDefault(require("./toast"));

exports.ToastPlugin = _toast.default;

var _tooltip = _interopRequireDefault(require("./tooltip"));

exports.TooltipPlugin = _tooltip.default;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
