// bootstrap-vue/es/components/card/index.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _card = _interopRequireDefault(require("./card"));

exports.BCard = _card.default;

var _cardHeader = _interopRequireDefault(require("./card-header"));

exports.BCardHeader = _cardHeader.default;

var _cardBody = _interopRequireDefault(require("./card-body"));

exports.BCardBody = _cardBody.default;

var _cardTitle = _interopRequireDefault(require("./card-title"));

exports.BCardTitle = _cardTitle.default;

var _cardSubTitle = _interopRequireDefault(require("./card-sub-title"));

exports.BCardSubTitle = _cardSubTitle.default;

var _cardFooter = _interopRequireDefault(require("./card-footer"));

exports.BCardFooter = _cardFooter.default;

var _cardImg = _interopRequireDefault(require("./card-img"));

exports.BCardImg = _cardImg.default;

var _cardImgLazy = _interopRequireDefault(require("./card-img-lazy"));

exports.BCardImgLazy = _cardImgLazy.default;

var _cardText = _interopRequireDefault(require("./card-text"));

exports.BCardText = _cardText.default;

var _cardGroup = _interopRequireDefault(require("./card-group"));

exports.BCardGroup = _cardGroup.default;

var _plugins = require("../../utils/plugins");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var components = {
  BCard: _card.default,
  BCardHeader: _cardHeader.default,
  BCardBody: _cardBody.default,
  BCardTitle: _cardTitle.default,
  BCardSubTitle: _cardSubTitle.default,
  BCardFooter: _cardFooter.default,
  BCardImg: _cardImg.default,
  BCardImgLazy: _cardImgLazy.default,
  BCardText: _cardText.default,
  BCardGroup: _cardGroup.default
};
var _default = {
  install: (0, _plugins.installFactory)({
    components: components
  })
};
exports.default = _default;
