// bootstrap-vue/es/components/form/index.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _form = _interopRequireDefault(require("./form"));

exports.BForm = _form.default;

var _formDatalist = _interopRequireDefault(require("./form-datalist"));

exports.BFormDatalist = _formDatalist.default;

var _formText = _interopRequireDefault(require("./form-text"));

exports.BFormText = _formText.default;

var _formInvalidFeedback = _interopRequireDefault(require("./form-invalid-feedback"));

exports.BFormInvalidFeedback = _formInvalidFeedback.default;

var _formValidFeedback = _interopRequireDefault(require("./form-valid-feedback"));

exports.BFormValidFeedback = _formValidFeedback.default;

var _formRow = _interopRequireDefault(require("../layout/form-row"));

var _plugins = require("../../utils/plugins");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var components = {
  BForm: _form.default,
  BFormDatalist: _formDatalist.default,
  BDatalist: _formDatalist.default,
  BFormText: _formText.default,
  BFormInvalidFeedback: _formInvalidFeedback.default,
  BFormFeedback: _formInvalidFeedback.default,
  BFormValidFeedback: _formValidFeedback.default,
  // Added here for convenience
  BFormRow: _formRow.default // BFormRow is not exported here as a named export, as it is exported by Layout

};
var _default = {
  install: (0, _plugins.installFactory)({
    components: components
  })
};
exports.default = _default;
