// bootstrap-vue/es/components/input-group/index.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _inputGroup = _interopRequireDefault(require("./input-group"));

exports.BInputGroup = _inputGroup.default;

var _inputGroupAddon = _interopRequireDefault(require("./input-group-addon"));

exports.BInputGroupAddon = _inputGroupAddon.default;

var _inputGroupPrepend = _interopRequireDefault(require("./input-group-prepend"));

exports.BInputGroupPrepend = _inputGroupPrepend.default;

var _inputGroupAppend = _interopRequireDefault(require("./input-group-append"));

exports.BInputGroupAppend = _inputGroupAppend.default;

var _inputGroupText = _interopRequireDefault(require("./input-group-text"));

exports.BInputGroupText = _inputGroupText.default;

var _plugins = require("../../utils/plugins");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var components = {
  BInputGroup: _inputGroup.default,
  BInputGroupAddon: _inputGroupAddon.default,
  BInputGroupPrepend: _inputGroupPrepend.default,
  BInputGroupAppend: _inputGroupAppend.default,
  BInputGroupText: _inputGroupText.default
};
var _default = {
  install: (0, _plugins.installFactory)({
    components: components
  })
};
exports.default = _default;
